security/auth.go

```diff
@@ -119,19 +130,16 @@ func AuthenticationHook(insecureMode bool, tlsState *tls.ConnectionState) (
 			return util.Errorf("missing User in request: %+v", request)
 		}
 
+		// If running in secure mode, the client certificate user must match the requested user.
```

Comment:
this is wrong. "node" should be able to act as anyone (we really want to allow forwarding from one node to another).
this should remain in the order it was before.